apps/remote_control/lib/lexical/remote_control/build/progress.ex

```diff
@@ -0,0 +1,16 @@
+defmodule Lexical.RemoteControl.Build.Progress do
+  alias Lexical.RemoteControl
+
+  import Lexical.RemoteControl.Api.Messages
+
+  def with_progress(label, func, opts \\ []) when is_function(func, 0) do
+    mix? = Keyword.get(opts, :mix?, true)
+    label = if mix?, do: "mix " <> label, else: label
+
+    RemoteControl.notify_listener(project_progress(label: label, stage: :begin))
+    result = func.()
+    RemoteControl.notify_listener(project_progress(label: label, stage: :end))
```

Comment:
you sure this shouldn't be wrapped with a try / after ? If func.() blows up for any reason, you'll never send the `end` message, despite the fact that the task has indeed ended (just not successfully)

```suggestion
try do
    RemoteControl.notify_listener(project_progress(label: label, stage: :begin))
    result = func.()
after
    RemoteControl.notify_listener(project_progress(label: label, stage: :end))
end
```